management/networksecuritygroup/client.go

```diff
@@ -0,0 +1,235 @@
+// Package networksecuritygroup implements operations for managing network security groups
+// using the Service Management REST API
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913824.aspx
+package networksecuritygroup
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/management"
+)
+
+const (
+	createSecurityGroupURL           = "services/networking/networksecuritygroups"
+	deleteSecurityGroupURL           = "services/networking/networksecuritygroups/%s"
+	getSecurityGroupURL              = "services/networking/networksecuritygroups/%s"
+	listSecurityGroupsURL            = "services/networking/networksecuritygroups"
+	addSecurityGroupToSubnetURL      = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups"
+	getSecurityGroupForSubnetURL     = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups"
+	removeSecurityGroupFromSubnetURL = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups/%s"
+	setSecurityGroupRuleURL          = "services/networksecuritygroups/%s/rules/%s"
+	deleteSecurityGroupRuleURL       = "services/networksecuritygroups/%s/rules/%s"
+
+	errParamNotSpecified = "Parameter %s is not specified."
+)
+
+// NewClient is used to instantiate a new SecurityGroupClient from an Azure client
+func NewClient(client management.Client) *SecurityGroupClient {
+	return &SecurityGroupClient{client: client}
+}
+
+// CreateNetworkSecurityGroup creates a new network security group within
+// the context of the specified subscription
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913818.aspx
+func (sg *SecurityGroupClient) CreateNetworkSecurityGroup(
+	name string,
```

Comment:
Why would you want/need that here? The arguments here are all used to create a new group and it are only 3 strings, so the signature is still pretty clean and very easy to use.

Or are you saying that you would like to switch the default approach for all functions to use parameter structs? So what would be the guideline/general rule for that? Only have one argument being a parameter struct? Of having a few parameters for the mandatory fields and all optional ones into a parameter struct? But what if you have 7 or 8 mandatory fields and the signature gets really long?

A lot of questions/options so some more details are welcome.